src/scm/gitlab.js

```diff
@@ -0,0 +1,278 @@
+    const changed = $('.diff-file:checked').toArray().map(elem => elem.value);
+    const changedFiles = changed.filter(f => code.gas[f]).map(f => {
+      return {name: f.replace(/\.gs$/, context.config.filetype), content: code.gas[f]}
+    });
+    const deleteFiles = changed.filter(f => !code.gas[f]);
+    const newFileNames = changed.filter(f => !code.scm[f]);
+    const updatedFileNames = changed.filter(f => !newFileNames.includes(f));
+
+    const newFiles = changedFiles.filter(f => newFileNames.includes(f.name));
+    const updatedFiles = changedFiles.filter(f => updatedFileNames.includes(f.name));
+
+    const comment = $('#commit-comment').val();
+
+    this.commitFiles(context.repo.fullName, context.branch, null, newFiles, updatedFiles, deleteFiles, comment)
+      .then(() => {
+        showAlert(`Successfully push to ${context.branch} of ${context.repo.fullName}`);
+      })
+      .catch((err) => {
+        showAlert('Failed to push', LEVEL_ERROR);
+      });
+  }
+
+  getAllBranches() {
+    context.repo.id = context.repo.id || this.namesToIds.repos[context.repo.fullName];
+    return getAllItems(Promise.resolve(
+      {
+        token: this.accessToken,
+        items: [],
+        url: `${this.baseUrl}/projects/${context.repo.id}/repository/branches?${this.tokenParam}`
+      }),
+      this.followPaginate,
+      'gitlab'
+    );
+  }
+
+  getCode() {
+    return new Promise((resolve, reject) => {
+      return $.getJSON(
+        `${this.baseUrl}/projects/${context.repo.id}/repository/tree?ref=${context.branch}&recursive=true&${this.tokenParam}`, {}
+      )
+        .then(resolve)
+        .fail(reject)
+    })
+        .then(response => {
+          const re = new RegExp(`(\\${context.config.filetype}|\\.html${context.config.manifestEnabled ? '|^appsscript.json' : ''})$`);
+          const promises = response.filter((tree) => {
+            return tree.type === 'blob' && re.test(tree.path);
+          })
+            .map(tree => {
+              var xx = `${this.baseUrl}/projects/${context.repo.id}/repository/files/${encodeURIComponent(tree.path)}?ref=${context.branch}&${this.tokenParam}`;
+              return new Promise((resolve, reject) => {
+                $.getJSON(xx, {})
+                  .then((content) => {
+                    resolve({file: tree.path, content: decodeURIComponent(escape(atob(content.content)))});
+                  })
+                  .fail(reject)
+              });
+            });
+          return Promise.all(promises);
+        });
+  }
+
+  getNamespaces() {
+    let testUrl = `${this.baseUrl}/groups?${this.tokenParam}`;
+    return getAllItems(Promise.resolve(
+      {
+        token: this.accessToken,
+        items: [],
+        url: testUrl
+      }),
+      this.followPaginate,
+      'gitlab'
+    )
+      .then(groups => {
+        this.namespaces = [this.user].concat(groups.map(group => group.name));
+        return this.namespaces;
+      })
+      .catch((err) => {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/leonhartX/gas-github/blob/5becb274ac17be98fee6e945e23cbab998e9d302/src/scm/gitlab.js#L156"></span>‘err’ is defined but never used.</td>
    </tr>
  </tbody>
</table>